spray-http/src/main/scala/spray/http/Uri.scala

```diff
@@ -330,18 +329,19 @@ object Uri {
   def httpScheme(securedConnection: Boolean = false) = if (securedConnection) "https" else "http"
 
   case class Authority(host: Host, port: Int = 0, userinfo: String = "") extends ToStringRenderable {
-    def isEmpty = host.isEmpty
-    def render[R <: Rendering](r: R): r.type = render(r, "", UTF8)
-    def render[R <: Rendering](r: R, scheme: String, charset: Charset): r.type =
-      if (isEmpty) r else {
+    def isEmpty = equals(Authority.Empty)
+    def nonEmpty = !isEmpty
+    def render[R <: Rendering](r: R): r.type = render(r, "", Path.Empty, UTF8)
+    def render[R <: Rendering](r: R, scheme: String, charset: Charset): r.type = render(r, scheme, Path.Empty, charset)
```

Comment:
Previously I deleted this render method (the one with 3 arguments: R, String and Charset) but now I have reverted it (for compatibility). Maybe someone is using it outside spray.